modules/creating-oci-compatible-connection-types-api.adoc

```diff
@@ -0,0 +1,147 @@
+
+.Prerequisites
+* You have created an OCI connection Secret in the `project` namespace.
+* You have deployed a KServe Operator in your cluster.
+
+.Procedure
+. Create a YAML file (for example, `oci-inferenceservice.yaml`) that defines the KServe `InferenceService` custom resource.
+. Specify the OCI connection annotation in the `metadata.annotations` section. Add the `opendatahub.io/connections` annotation and set its value to reference the OCI Secret name, `my-oci-connection`.
+. Define the model format by configuring the `.spec.predictor.model.modelFormat` field to specify the framework of the model (for example, `pytorch`).
++
+[source,bash]
+----
+apiVersion: serving.kserve.io/v1beta1
+kind: InferenceService
+metadata:
+  name: my-model                   # Name of the service
+  namespace: my-project
+  annotations:
+    opendatahub.io/connections: 'my-oci-connection'      # Reference to the OCI Connection Secret
+spec:
+  predictor:
+    model:
+      modelFormat:
+        name: pytorch             # Specify the framework format (for example, pytorch)
+      # NOTE: The operator webhook creates and injects .spec.predictor.imagePullSecrets
+      # for OCI authentication based on the Secret.
+----
+
+. Apply the `InferenceService` custom reource by using the `kubectl apply` command to create the resource.
++
+[source,bash]
+----
+kubectl apply -f oci-inferenceservice.yaml
+----
+
+== Using an OCI connection with `LLMInferenceService` custom resource
+You can use an OCI-compatible connection type with the `LLMInferenceService` custom resource. In the following procedure, you define the private image registry location for your Large Language Model (LLM) container image by using an OCI connection when deploying a KServe `LLMInferenceService`.
+
+.Prerequisites
+* You have created an OCI connection Secret in the `project` namespace. For more information, see _Creating an OCI connection type using the Connections API_.
+* You have a KServe Operator deployed that supports the `LLMInferenceService` custom resource.
+
+.Procedure
+. Create a YAML file (for example, `oci-llm-service.yaml`) that defines the KServe `LLMInferenceService` custom resource.
+. Specify the OCI connection annotation in the `metadata.annotations` section. Add the `opendatahub.io/connections` annotation and set its value to reference the OCI Secret name, `my-oci-connection`. 
++
+[source,yaml]
+----
+apiVersion: serving.kserve.io/v1alpha1
+kind: LLMInferenceService
+metadata:
+  name: my-llm-model                   # Name of the LLM serving instance
+  namespace: my-project
+  annotations:
+    opendatahub.io/connections: 'my-oci-connection'      # Reference to the OCI Connection Secret
+spec:
+  model:
+    # Define the container image path here, if required.
+    # NOTE: The operator webhook automatically injects imagePullSecrets
```

Comment:
```suggestion
    # NOTE: The operator webhook automatically injects .spec.template.imagePullSecrets
```
This is the path that the webhook will inject OCI data to for `LLMInferenceSerivces`